### DISCUSSION TITLE
2 different yml file at root of repository

### DISCUSSION THREAD

**Discussion by tranvutuan** [SwiftGen/SwiftGen#845] [Q&A] [answered]
Im curious if we can put 2 different yml files at the same root level so that we can load a different config for each target.
For example:
```
target A would use swiftgen1.yml
  --- images
  --- localization
target B would use swiftgen1.yml
  --- images
  --- localization
target Shared ( which is shared between A and B ) would use swiftgen2.yml
 --- images
 --- localization
```

**Comment by djbe** [comment, marked as answer]
Nothing is stopping you from doing that. We specifically have a CLI option to provide the path to a config file:
```bash
swiftgen config run --config=swiftgen1.yml
swiftgen config run --config=swiftgen2.yml
```

Just make sure that the output paths (in each config) are different, otherwise the commands will overwrite each other.

**Reply by tranvutuan** [reply]
quick question. we have to use CLI right ? since I was trying to load different yml files ( like I mention on OP post ) via `New Run Script Phase` but getting an error during building such as can't not regconize the path for `swiftgen1.yml` when building `shared` target ? unless I have to change `swiftgen1.yml` to `swiftgen.yml` ( `swiftgen2` is still as is for `shared` target )

**Reply by djbe** [reply]
Your run script phase is just going to run some commands in your project's directory, so your paths need to be relative to that.

Honestly, seeing as you have 3 targets, I'd just have 3 separate configs. Even though the 2 configs of A and B may be equal in contents, it'd be good practice to split them up. That way if you ever modify target A a bit, it won't impact stuff for target B.

If your folder structure is (for example):
- MyProject Root
  - MyProject.xcodeproj
  - A/
    - swiftgen.yml
    - Images/...
    - en.lproj/...
  - B/
    - swiftgen.yml
    - Images/...
    - en.lproj/...
  - Shared/
    - swiftgen.yml
    - Images/...
    - en.lproj/...

### Solution 1

You'll have a swiftgen.yml configuration file for each target, **inside** that target's directory. All the paths **in your config** will need to include the target's folder path, so prefix everything with `A/` (or B or Shared), or set the `input_dir` and `output_dir` to something relative to (and including) `A/`.

To test these (outside Xcode), just run `swiftgen config run --config=A/swiftgen.yml` (and so on). Once you verify this works for **each** target's config, you can add the build phases.

For each target (A, B, and Shared), add a build phase to run SwiftGen, as documented [here](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/Articles/Xcode-Integration.md). A small difference will be that your path to the config will be in that target's folder, so the command could for example be:

```bash
swiftgen config run --config=A/swiftgen.yml
```

### Solution 2

Similar to 1, but the paths in your config **do not** include the target's folder, only the sub-paths. This allows you to simplify the config files a little bit.

To test the config files, `cd` to each target's folder, and then run `swiftgen config run`. Once you verify this works for each target's folder, you can add the build phases.

Your build phases will now be slightly different, i.e. you need to change the working directory to the target's folder:
```bash
pushd "A"
swiftgen config run
popd
```

**Reply by tranvutuan** [reply]
Thanks! and let me give it a try and get back to you.

**Reply by tranvutuan** [reply]
@djbe sorry for super late response since I have been kept busy on few things. I tried to use swift config run and it worked.
